src/components/sync-connected-sites.tsx

```diff
@@ -118,14 +121,18 @@ const SyncConnectedSitesSection = ( {
 
 	const handlePushSite = async ( connectedSite: SyncSite ) => {
 		if ( connectedSite.isStaging ) {
-			showPushStagingConfirmation( () => {
-				updateTimestamp( selectedSite.id, connectedSite.id, 'push' );
-				pushSite( connectedSite, selectedSite );
+			showPushStagingConfirmation( async () => {
+				const success = await pushSite( connectedSite, selectedSite );
+				if ( success ) {
+					await refreshTooltip( selectedSite.id, connectedSite.id, 'push' );
+				}
```

Comment:
Why not executing this code inside the "pushSite" ?
The same comment for pullSite.